src/scenic/core/scenarios.py

```diff
@@ -331,7 +385,7 @@ def generate(self, maxIterations=2000, verbosity=0, feedback=None):
                 sample = Samplable.sampleAll(self.dependencies)
             except RejectionException as e:
                 optionallyDebugRejection(e)
-                rejection = e
+                rejection = f"Rejection exception: {e}"
```

Comment:
"Rejection exception" isn't meaningful to ordinary users, so I'd just keep this as `e` (the string form of the exception is supposed to be an explanatory string).